base/cmd.jl

```diff
@@ -242,6 +243,29 @@ setenv(cmd::Cmd, env::Pair{<:AbstractString}...; dir="") =
     setenv(cmd, env; dir=dir)
 setenv(cmd::Cmd; dir="") = Cmd(cmd; dir=dir)
 
+"""
+    addenv(command::Cmd, env...)
+
+Merge new environment mappings into the given `Cmd` object, returning a new `Cmd` object.
+Duplicate keys are replaced.
+"""
+function addenv(cmd::Cmd, env::Pair{<:AbstractString}...)
```

Comment:
Maybe make the core method accept `Dict` just like `setenv`
```suggestion
function addenv(cmd::Cmd, env::Dict)
```